### ISSUE TITLE
Support for containers ("open in new tab", more?)

### ISSUE THREAD

**Issue by Reylak** [Robbendebiene/Gesturefy#234]
For some time, Firefox has had support for [tab containers](https://addons.mozilla.org/fr/firefox/addon/multi-account-containers/) under the form of a plugin. Basically, this allows one to isolate tabs from each other, which includes cookies and local storage. A good thing for privacy in general.

My suggestion is to add support for containers in Gesturefy, so e.g. you can assign a gesture to opening a link to a new tab inside a designated container. Maybe some other gestures or actions may benefit from container support.

I know that it is possible for plugins to interact with containers right now, see for instance [Context Plus](https://addons.mozilla.org/en-US/firefox/addon/context-plus/) or [Cookie Autodelete](https://addons.mozilla.org/en-US/firefox/addon/cookie-autodelete/).

From an implementation point of view, I'd have a special button beside the "Open link" entry (and any relevant entry), like a plus sign with the mention of "containers". Clicking on it would add a new entry to enter a gesture. Under it is a drop-down menu from where to chose a container.

Display of this special button would be conditioned to the availability of containers, and maybe to a special setting such as "Enable container support" as Cookie Autodelete does.

I understand this is a somewhat advanced feature that relies on a feature that is itself not even implemented as a default in Firefox. It's just an idea passing by :)

**Comment by Robbendebiene** [comment]
Hey, thanks for your feature request.
I recently also started enjoying the new tab containers, so I'm pretty much aware of its benefits and usefulness :)

> From an implementation point of view, I'd have a special button beside the "Open link" entry (and any relevant entry), like a plus sign with the mention of "containers". Clicking on it would add a new entry to enter a gesture. Under it is a drop-down menu from where to chose a container.

Do you just want an option (like an advanced setting) where you can choose in which container the tab will be opened? I'm asking this because of:

> Clicking on it would add a new entry to enter a gesture.

which sounds similar to this https://github.com/Robbendebiene/Gesturefy/issues/197 or at least could be covered by it. 

I also could imagine a popup action (upcoming feature) where you can select the container in which the url should be opened.

**Comment by Reylak** [comment]
Yes this looks exactly like #197 actually! Add a new action of type "Open in a new tab" or whatever, with an attached advanced setting to choose the container. Popup action is interesting too, maybe better because you would only have one action and then the user dynamically chooses which container to open the tab in.

**Comment by Robbendebiene** [comment]
While developing version 2.0 I looked into this and came across some problems. The main concern is that this feature requires at least 2 new permissions where one of them cannot be declared as optional. Since I tried minimizing the amount of key permissions Gesturefy needs, I won't implement this feature yet. Nevertheless I still like the idea and can imagine some good use cases, so I'll keep this issue open as a reminder.

**Comment by cprn** [comment]
There's an [add-on that can open tabs in containers](https://github.com/honsiorovskyi/open-url-in-container/) by adding a custom protocol extension. I'm pretty sure it can be used with Gesturefy's "Send message to other add-on" or "Execute user-script" actions but idk how. Made a [ticket on their issue tracker](https://github.com/honsiorovskyi/open-url-in-container/issues/105) as well.

**Comment by Robbendebiene** [comment]
@cprn  The "Send message to other add-on" command won't work here since you currently cannot get any data from the gesture target (like the link url) wherefore you cannot pass the data to another add-on. However a user script can access the target but opening a **new tab** via java script will most likely not work the way you want.

**Comment by Robbendebiene** [comment]
Just for clarification why this issue hasn't been moved forward: I'm still waiting for https://bugzilla.mozilla.org/show_bug.cgi?id=1386673 to be resolved.

**Comment by adenlall** [comment]
Is there any advancing in this issue

**Comment by Robbendebiene** [comment]
> Just for clarification why this issue hasn't been moved forward: I'm still waiting for https://bugzilla.mozilla.org/show_bug.cgi?id=1386673 to be resolved.

Still applies

**Comment by cprn** [comment]
I'm leaving it here because I won't be working on it after all but I think it's an idea worthy exploring - "open in container" seems like a perfect candidate.

We've had menus open after a gesture for quite some time (e.g. "restore tab" can display history of recently closed tabs to choose from) but we don't get them as gesture controlled menus, e.g. radial ones with 8 default directional gestures. This way it'd be possible to employ whole gesture sequences, e.g. gesture down-up for "duplicate tab", then release the mouse button, radial menu with containers wildly appears, press the mouse button again and gesture right for "shopping" container, and so on.

Just some thoughts. Sorry if that's a SPAM for some. Will shut up now. Cheers! 🥂

**Comment by Dituar** [comment]
Add the "Open link in a container (pre-selected) in a new tab" gesture and go to that tab. Switch to the tab is "On/Off" option.
Add the "Open link in a container (popup for your choice) in a new tab and go to that tab" gesture. Switch to the tab is "On/Off" option.

I have FoxyProxy configured with specific proxyes for specific containers. It would be great to be able to use gestures to open links in the required containers (to open in the required proxy).

**Comment by Reylak** [comment]
Firefox's bug "Make Contextual Identity extensions be an optional permission" still applies I guess. Plus there's this one about having this feature without asking for a new permission, but read-only: https://bugzilla.mozilla.org/show_bug.cgi?id=1947602 "Allow limited read-only access to contextualIdentities.query without permission".

The whole problem is about accessing the list of containers without asking for a new permission, and even before that, about implementing optional container usage in Gesturefy without automatically enabling containers to people that do not want them.
